acados/ocp_nlp/ocp_nlp_cost_nls.c

```diff
@@ -784,7 +784,7 @@ void ocp_nlp_cost_nls_update_qp_matrices(void *config_, void *dims_, void *model
 
         // RSQrq = RSQrq + tmp_nv_ny * tmp_nv_ny^T
         blasfeo_dsyrk_ln(nu+nx, ny, model->scaling, &work->tmp_nv_ny, 0, 0, &work->tmp_nv_ny, 0, 0,
-                         model->scaling, &work->tmp_nv_nv, 0, 0, memory->RSQrq, 0, 0);
+                         1.0, memory->RSQrq, 0, 0, memory->RSQrq, 0, 0);
```

Comment:
It was correct as it was before, since the gauss-newton term was added to (the scaled version of) the term `hess(r)*r`, which was stored in `tmp_nv_nv`. After your "fix", this term is disregarded and the code computes the same as in the `opts->gauss_newton_hess` case.

Reply:
If anything, the comment just above is to be fixed.

Reply:
Thanks a lot for this comment!
I will undo this change and fix the comment above.

Moreover, I will have a look at the other cost modules, I made this change to make the NLS cost equivalent to the other two on the test problem.
Which makes me think now, that the other two cost modules do have the bug that I introduced here..